test/delayP.js

```diff
@@ -0,0 +1,41 @@
+import * as RA from '../src';
+import eq from './shared/eq';
+
+describe('delayP', function() {
```

Comment:
Cool, let's now deal with the tests. First think we need to do is to aligned the style of the tests to new conventions. These conventions can be seen for example here: https://github.com/char0n/ramda-adjunct/blob/a56484defec103cd1bcff2f55099b9699d876892/test/allP.js#L5-L4. We're working hard on migrating all our tests to this style. Another thing is to use explic asserts instead of internal `eq` function.
```js
describe('delayP', function() {
  context('given an number as an argument', function() {
    specify('should delay the resolution of the returned promise', async function() {
       const delayedValue = await RA.delayP(10);
       
       assert.isUndefined(delayedValue);
    });
  });
});
```
Notice we can use power or mocha framework to write asynchronous tests directly usgin `await`. This is example how to write async test for rejections. Also don't use big numbers for delays - big numbers makes our entire suite slower by this delay.
```js
describe('delayP', function() {
  context('reject', function() {
    context('given an number as an argument', function() {
      specify('should delay the rejected of the returned promise', async function() {
        try {
          const delayedValue = await RA.delayP.reject(10);
          throw new Error('rejection should fail');
        } catch (e) {       
          assert.isUndefined(delayedValue);
        }
      });
    });
  });
});
```

`test delay when passing object with different properties as argument` - given object with unknown properties -> should use defaults.

We're also completely missing tests for currying. 

Let me know if you have any questions.

Reply:
What test for currying we need to work on?

Reply:
Here is an example of unary functional currying test: https://github.com/char0n/ramda-adjunct/blob/22f4ec2f3b9917391959c96a3ffb36381c82ae17/test/isNilOrEmpty.js#L97-L96

Reply:
Can you quickly tell me how to test for equality in error objects. I am currently trying something like this but it seems `chai` does not support `deepEqual()` on error objects:
```js
    context('passing object as argument', function() {
      specify(
        'should delay the rejected of the returned promise',
        async function() {
         try {
            const delayedValue = RA.delayP.reject({
              timeout: 900,
              value: new Error('Gracefull Error'),
            });
            throw new Error('rejection should fail');
          } catch (e) {
            assert.deepEqual(new Error('Gracefull Error'), delayedValue);
          }
        }
      );
    });
```

Reply:
Can you help me with the above one

Reply:
Sorry for delay in answer, I took some time off. Ok so problem is assert of errors and equality. In your case it's sufficient to do strict equality comparison:
```js
    context('passing object as argument', function() {
      specify(
        'should delay rejection of the returned promise',
        async function() {
        const error = new Error('error');

         try {
            const actual = await RA.delayP.reject({
              timeout: 90,
              value: error,
            });
            throw new Error('rejection should fail');
          } catch (e) {
            assert.strictEqual(actual, error);
          }
        }
      );
    });